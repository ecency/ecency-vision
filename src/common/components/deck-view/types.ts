export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
