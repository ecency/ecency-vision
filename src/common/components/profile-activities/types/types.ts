export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }