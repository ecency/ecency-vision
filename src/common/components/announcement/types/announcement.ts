export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
