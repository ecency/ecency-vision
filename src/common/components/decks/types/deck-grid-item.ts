interface BaseDeckGridItem {
  id: string;
  key: number;
  type:
    | "u"
    | "ac"
    | "co"
    | "w"
    | "n"
    | "tr"
    | "to"
    | "s"
    | "cu"
    | "th"
    | "msf"
    | "faq"
    | "wb"
    | "wn";
  settings: Record<string, any>;
}

export interface ReloadableDeckGridItem extends BaseDeckGridItem {
  settings: {
    updateIntervalMs: number;
  };
}

export interface UserDeckGridItem extends ReloadableDeckGridItem {
  type: "u";
  settings: ReloadableDeckGridItem["settings"] & {
    username: string;
    contentType: string;
  };
}

export interface CommunityDeckGridItem extends ReloadableDeckGridItem {
  type: "co";
  settings: UserDeckGridItem["settings"] & {
    tag: string;
  };
}

export interface SearchDeckGridItem extends ReloadableDeckGridItem {
  type: "s";
  settings: ReloadableDeckGridItem["settings"] & {
    query: string;
    author?: string;
    type?: string;
    category?: string;
    tags?: string;
    date?: string;
    sort?: string;
    hideLow?: boolean;
  };
}

export interface WavesDeckGridItem extends ReloadableDeckGridItem {
  type: "th";
  settings: ReloadableDeckGridItem["settings"] & {
    host: string;
  };
}

export type DeckGridItem =
  | BaseDeckGridItem
  | UserDeckGridItem
  | CommunityDeckGridItem
  | SearchDeckGridItem
  | WavesDeckGridItem;
