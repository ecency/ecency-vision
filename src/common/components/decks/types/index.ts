export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
