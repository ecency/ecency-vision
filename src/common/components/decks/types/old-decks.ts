interface GenericDeck<DF extends Record<string, string | number> | null> {
  createdAt: string;
  dataFilters: DF;
  dataParams: string[];
  header: {
    title: string;
    updateIntervalMs: number;
  };
}

export interface OldNotificationDeck extends GenericDeck<{ type: string; group: string }> {}
export interface OldNonDFDeck extends GenericDeck<null> {}

export type OldDeck = OldNotificationDeck | OldNonDFDeck;
