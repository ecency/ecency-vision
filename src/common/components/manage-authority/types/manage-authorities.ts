import { Account, FullAccount } from "../../../store/accounts/types";
import { PublicKey } from "@hiveio/dhive";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting",
  Memo: "memo"
};

export interface PublicKeys {
  publicOwnerKey?: string;
  publicActiveKey?: string;
  publicPostingKey?: string;
  publicMemoKey?: string;
}

export interface AccountDataType {
  postingsAuthority: [string, number][];
  posting: [string | PublicKey, number];
  owner: [string | PublicKey, number];
  active: [string | PublicKey, number];
  weight: number;
  memokey: string;
  account: Account;
  publicKeys: {
    publicOwnerKey: string | PublicKey;
    publicActiveKey: string | PublicKey;
    publicPostingKey: string | PublicKey;
    publicMemoKey: string;
  };
}

export const actionType = {
  Revoke: "revoke",
  Keys: "keys",
  Import: "import",
  Reveal: "reveal"
};
