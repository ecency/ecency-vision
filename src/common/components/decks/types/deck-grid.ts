import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  key: string;
  title: string;
  icon?: string;
  columns: DeckGridItem[];
  storageType: "account" | "local";
}

export interface DeckGrids {
  decks: DeckGrid[];
}
