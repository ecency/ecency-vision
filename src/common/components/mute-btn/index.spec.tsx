import React from 'react';

import {MuteBtn, DialogBody} from './index';
import TestRenderer from 'react-test-renderer';

import {
  entryInstance1,
  communityInstance1,
  activeUserMaker,
} from '../../helper/test-helper';
import {Entry, EntryStat} from '../../store/entries/types';

const defProps = {
  entry: {...entryInstance1},
  community: {...communityInstance1},
  activeUser: activeUserMaker('foo'),
  onSuccess: () => {},
};

it("(1) Should show 'mute' label", () => {
  const props = {...defProps};
  const renderer = TestRenderer.create(<MuteBtn {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(2) Should show 'unmute' label", () => {
  const stats = {...entryInstance1.stats!, gray: true};
  const entry = {...entryInstance1, stats};

  const props = {...defProps, entry: entry};
  const renderer = TestRenderer.create(<MuteBtn {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(3) Dialog body for 'mute'", () => {
  const props = {
    entry: {...entryInstance1},
    inProgress: false,
    onSubmit: () => {},
  };
  const renderer = TestRenderer.create(<DialogBody {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(4) Dialog body for 'unmute'", () => {
  const stats: EntryStat = {
    hide: false,
    gray: true,
    total_votes: 10,
    flag_weight: 1.0,
  };

  const props = {
    entry: {
      ...entryInstance1,
      stats,
    },
    inProgress: false,
    onSubmit: () => {},
  };
  const renderer = TestRenderer.create(<DialogBody {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});
