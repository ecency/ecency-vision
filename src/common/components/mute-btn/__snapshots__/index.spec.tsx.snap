// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Should show 'mute' label 1`] = `
<a
  className="mute-btn"
  href="#"
  onClick={[Function]}
>
  Mute
</a>
`;

exports[`(2) Should show 'unmute' label 1`] = `
<a
  className="mute-btn"
  href="#"
  onClick={[Function]}
>
  Unmute
</a>
`;

exports[`(3) Dialog body for 'mute' 1`] = `
<div
  className="mute-form"
>
  <div
    className="mb-4"
  >
    <div
      className="entry-title"
    >
      @
      good-karma
      /
      awesome-hive
    </div>
    <div
      className="ecency-input-group flex items-stretch w-full [&>input]:rounded-[0] [&>input:first-child]:rounded-l-full [&>input:last-child]:rounded-r-full [&>.ecency-input-group-part>button]:rounded-tl-none [&>.ecency-input-group-part>button]:rounded-bl-none [&>.ecency-input-group-part>button]:h-[2.75rem]"
    >
      <input
        autoComplete="off"
        autoFocus={true}
        className="border-2 rounded-3xl py-2 px-3 w-full outline-none shadow-0 focus:border-gray-500 hover:border-gray-300 duration-300 dark:border-gray-700 dark:hover:border-gray-600 dark:bg-gray-800 aria-invalid:border-red"
        maxLength={120}
        onChange={[Function]}
        placeholder="Notes"
        type="text"
        value=""
      />
    </div>
    <small>
      Provide a note regarding your decision to mute this content.
    </small>
  </div>
  <div>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Mute
      </div>
    </button>
  </div>
</div>
`;

exports[`(4) Dialog body for 'unmute' 1`] = `
<div
  className="mute-form"
>
  <div
    className="mb-4"
  >
    <div
      className="entry-title"
    >
      @
      good-karma
      /
      awesome-hive
    </div>
    <div
      className="ecency-input-group flex items-stretch w-full [&>input]:rounded-[0] [&>input:first-child]:rounded-l-full [&>input:last-child]:rounded-r-full [&>.ecency-input-group-part>button]:rounded-tl-none [&>.ecency-input-group-part>button]:rounded-bl-none [&>.ecency-input-group-part>button]:h-[2.75rem]"
    >
      <input
        autoComplete="off"
        autoFocus={true}
        className="border-2 rounded-3xl py-2 px-3 w-full outline-none shadow-0 focus:border-gray-500 hover:border-gray-300 duration-300 dark:border-gray-700 dark:hover:border-gray-600 dark:bg-gray-800 aria-invalid:border-red"
        maxLength={120}
        onChange={[Function]}
        placeholder="Notes"
        type="text"
        value=""
      />
    </div>
    <small>
      Provide a note regarding your decision to unmute this content.
    </small>
  </div>
  <div>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Unmute
      </div>
    </button>
  </div>
</div>
`;
